src/App.tsx

```diff
@@ -1,61 +1,119 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number) {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const newTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+let maxId = Math.max(...newTodos.map(todo => todo.id));
+
+export const App: React.FC = () => {
+  const [activeUser, setActiveUser] = useState(0);
+  const [message, setMessage] = useState('');
+  const [visibleTodos, setVisibleTodos] = useState(newTodos);
+  const [hasMessageError, setHasMessageError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const newTodo = () => {
+    maxId += 1;
+
+    return {
+      id: maxId,
+      title: message,
+      completed: false,
+      userId: activeUser,
+      user: getUser(activeUser),
+    };
+  };
+
+  const handleAdd = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setHasMessageError(!message);
+    setHasUserError(!activeUser);
```

Comment:
You are setting them here. What happens if I start to write in the input field and don't click submit?

Reply:
I really do not understand the comment, may you share the solution please ?

Reply:
[screenshot](https://dsc.cloud/2f6ec4/Screenshot-2023-01-27-at-15.21.01.png)

Reply:
You only set error on submit but you don't remove to error until it is submitted again.